.github/workflows/ci.yml

```diff
@@ -39,7 +73,17 @@ jobs:
             git diff
             echo "::endgroup::"
 
-            echo 'Changes to reference found, please regenerate with `npm run generate-reference-all` or `npm run generate-reference <specific directory>`'
+            cat <<EOF
+          Error: the references have differences between what's committed and what would be
+          generated. Either:
+
+          - if you made changes to the generation scripts, please regenerate with `npm run
+            generate-reference-all` or `npm run generate-reference <specific directory>`.
+
+          - if you have edited the files directly, https://github.com/pantsbuild/pants is the
+            source of truth: make the changes there and they will be synced here on the next
+            release.
+          EOF
```

Comment:
The tabbing of this seems off. Is this right?

Reply:
Yes, a heredoc doesn't care about the indentation of the command that starts it, the content and the ending `EOF` need to start in the first column of the script. This particular instance is extra weird because of the YAML layer. I'll switch this to a proper script like you suggest, and it'll look better.